src/plugins/data/common/index_patterns/types.ts

```diff
@@ -55,6 +68,9 @@ export interface IndexPatternAttributes {
   allowNoIndex?: boolean;
 }
 
+/**
+ * Storage of field attributes. Necessary since the field list isn't saved.
```

Comment:
I would mark this as @internal as it's not exported on the contracts of public \ server.